datahub-actions/src/datahub_actions/plugin/action/metadata_change_sync/metadata_change_sync.py

```diff
@@ -0,0 +1,112 @@
+import json
+import logging
+from typing import Dict, List, Optional, Set, Union, cast
+
+from datahub.emitter.rest_emitter import DatahubRestEmitter
+from datahub.metadata.schema_classes import (
+    MetadataChangeLogClass,
+    MetadataChangeProposalClass,
+)
+from pydantic import BaseModel
+
+from datahub_actions.action.action import Action
+from datahub_actions.event.event_envelope import EventEnvelope
+from datahub_actions.event.event_registry import METADATA_CHANGE_LOG_EVENT_V1_TYPE
+from datahub_actions.pipeline.pipeline_context import PipelineContext
+
+logger = logging.getLogger(__name__)
+
+
+class MetadataChangeEmitterConfig(BaseModel):
+    gms_server: Optional[str]
+    gms_auth_token: Optional[str]
+    aspects_to_exclude: Optional[List]
+    extra_headers: Optional[Dict[str, str]]
+
+
+class MetadataChangeSyncAction(Action):
+    rest_emitter: DatahubRestEmitter
+    aspects_exclude_set: Set
+    # By default, we exclude the following aspects since different datahub instances have their own encryption keys for
+    # encrypting tokens and secrets, we can't decrypt them even if these values sync to another datahub instance
+    # also, we don't sync execution request aspects because the ingestion recipe might contain datahub secret
+    # that another datahub instance could not decrypt
+    DEFAULT_ASPECTS_EXCLUDE_SET = {
+        "dataHubAccessTokenInfo",
+        "dataHubAccessTokenKey",
+        "dataHubSecretKey",
+        "dataHubSecretValue",
+        "dataHubExecutionRequestInput",
+        "dataHubExecutionRequestKey",
+        "dataHubExecutionRequestResult",
```

Comment:
@jjoyce0510 - we really need a notion of "internal" vs "external" entities/aspects